datalad/distribution/remove.py

```diff
@@ -270,15 +276,18 @@ def __call__(
         if not to_save:
             # nothing left to do, potentially all errored before
             return
+        if not save:
+            lgr.debug('Not calling `save` as instructed')
+            return
 
         for res in Save.__call__(
                 # TODO compose hand-selected annotated paths
                 path=to_save,
                 # we might have removed the reference dataset by now, recheck
-                dataset=refds_path if GitRepo.is_valid_repo(refds_path) else None,
-                # TODO allow for custom message
-                #message=message if message else '[DATALAD] removed content',
-                message='[DATALAD] removed content',
+                dataset=refds_path
+                        if (refds_path and GitRepo.is_valid_repo(refds_path))
+                        else None,
```

Comment:
To solve it that way doesn't really make sense to me. Judging from #1821 the determination of the reference datasets fails for some reason and needs to be fixed. It's not about having none. You can't `datalad remove` anything from a non-repository, right?

Reply:
well... I am not strong in all this logic... looking at original code though it even allowed refds_path point to some invalid repo as far as I see it ;)  so checking if it is defined at all, sounds legit to me.  moreover we could fix up instead is_valid_repo to not puke upon none?

Reply:
BTW, I did add a test for this usecase, so if you see a better/proper fix, you could check it out quickly.

Reply:
Either I have misread the diff or sth else was merged in the meantime. Whatever is the case - what I was unhappy about isn't valid (anymore).